src/ExportMap.ts

```diff
@@ -30,12 +30,12 @@ export class ExportMap {
 
   exportNested(scope: string, fileDescriptor: FileDescriptorProto, message: DescriptorProto) {
     const messageEntry: ExportMessageEntry = {
-      pkg: fileDescriptor.getPackage(),
-      fileName: fileDescriptor.getName(),
-      messageOptions: message.getOptions(),
-      mapFieldOptions: message.getOptions() && message.getOptions().getMapEntry() ? {
-        key: [message.getFieldList()[0].getType(), message.getFieldList()[0].getTypeName().slice(1)],
-        value: [message.getFieldList()[1].getType(), message.getFieldList()[1].getTypeName().slice(1)],
+      pkg: fileDescriptor.getPackage()!,
+      fileName: fileDescriptor.getName()!,
+      messageOptions: message.getOptions()!,
+      mapFieldOptions: message.getOptions() && message.getOptions()!.getMapEntry() ? {
+        key: [message.getFieldList()[0].getType()!, message.getFieldList()[0].getTypeName()!.slice(1)],
+        value: [message.getFieldList()[1].getType()!, message.getFieldList()[1].getTypeName()!.slice(1)],
```

Comment:
I'm assuming the API has changed to allow for `undefined` return types. Are we confident all these nullsafe operators won't bite us? Might be safer reflect the `undefined` type in the `ExportMessageEntry` type.

Reply:
I decided to redo this and added assertions using a `throwError` util function to avoid the explosion of separate checks. Can you give it a quick look to see what you think please?

Reply:
LGTM! Thanks for addressing.